### DISCUSSION TITLE
Randomlly sending tweets

### DISCUSSION THREAD

**Discussion by Ghonghito** [tweepy/tweepy#1478] [Questions] [answered]
Hello, im using tweepy 3.9,0 version and i have a bot which sends tweets to me in Telegram... my issue is when I'm streaming tweets from my account working fine every tweets is sending to me BUT when I'm streaming tweets from pages like @Binance sending tweets randomly even splitting text and spamming my chat like this and i can't get it why (full code will be bellow):

![myimg](https://i.imgur.com/IFcBKNr.png)

here is my full code: 

`import json
import tweepy
import telebot
from telebot import types
from telebot.types import Message
import logging
import time
import json

bot_token = 'TOKEN'

logger = telebot.logger
telebot.logger.setLevel(logging.DEBUG)
bot = telebot.TeleBot(token=bot_token)

logging.basicConfig(level=logging.DEBUG,
                    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')

auth = tweepy.OAuthHandler('TOKEN', 'TOKEN')
auth.set_access_token('TOKEN', 'TOKEN')
api = tweepy.API(auth)

class MyStreamListener(tweepy.StreamListener):
    
    def on_status(self, status):
        cid = "MY CHAT_ID IN TELEGRAM"
        bot.send_message(cid, status.text)
        
    def on_error(self, status_code):
        if status_code == 420:
            #returning False in on_error disconnects the stream
            return False
        
        
myStreamListener = MyStreamListener()
myStream = tweepy.Stream(auth = api.auth, listener=myStreamListener)       
myStream.filter(follow=["877807935493033984"])        
        
while True:
    try:
        bot.polling()
        break
    except Exception:
        time.sleep(30)
`

**Comment by Harmon758** [comment, marked as answer]
When specifying user IDs using the `follow` parameter of the [POST statuses/filter endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/filter-realtime/api-reference/post-statuses-filter) that `Stream.filter` uses:

> For each user specified, the stream will contain:
> 
> - Tweets created by the user.
> - Tweets which are retweeted by the user.
> - Replies to any Tweet created by the user.
> - Retweets of any Tweet created by the user.
> - Manual replies, created without pressing a reply button (e.g. “@​twitterapi I agree”).

https://developer.twitter.com/en/docs/twitter-api/v1/tweets/filter-realtime/guides/basic-stream-parameters

For code block usage, see https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/creating-and-highlighting-code-blocks.
